### PR TITLE
feat: add interactive credential mapping for promote

### LINKED ISSUE #490: FEATURE: Interactive credential mapping wizard for unresolvable credentials during promotion
**What would you like?**

When `n8nac promote` encounters credentials that cannot be resolved in the target environment, instead of blocking the promotion with an error, the CLI should offer an **interactive credential mapping wizard**.

The wizard would:

1. Detect all unresolvable credentials grouped by type (e.g. `postgres`, `httpBasicAuth`)
2. Query the target environment's available credentials of the matching type via the n8n API
3. Present an interactive prompt (e.g. using `inquirer` or similar) listing the available target credentials for each unresolvable source credential
4. Allow the user to either map the source credential to an existing target credential, skip it, or abort
5. Persist all confirmed mappings in `n8nac-promotion.json` under `routes["<route>"].bindings.credentials` — keyed as `<name>:<type>` for both source and target — so future promotions are non-interactive and fully repeatable

The mappings integrate naturally into the existing promotion schema:

```json
{
  "version": 2,
  "routes": {
    "test->prod": {
      "bindings": {
        "workflows": {},
        "credentials": {
          "<source-credential-name>:<credential-type>": "<target-credential-name>:<credential-type>"
        }
      },
      "workflowOverrides": {},
      "credentialOverrides": {},
      "nameRules": []
    }
  }
}
```

A `--no-interactive` flag should suppress the wizard and keep the current blocking behavior for CI/CD pipelines.

---

**Why would it be useful?**

Currently, any credential name mismatch between environments causes the entire promotion to be **blocked**, even if most workflows are perfectly promotable. Users must manually identify which credentials exist in the target, edit `n8nac-promotion.json` by hand, and re-run the promotion — with no guidance on what target credentials are actually available.

This is especially painful when promoting across environments where credentials intentionally have different names (e.g. a test database credential mapped to its production counterpart). The mapping is a one-time decision that should be captured in version-controlled config, not re-derived on every run.

The wizard would turn a hard blocker into a guided, self-documenting setup step that writes directly into the established `bindings.credentials` structure, making `n8nac promote` viable for teams onboarding new environments without deep knowledge of the target's credential inventory.

### PR DESCRIPTION
**Pull request by EtienneLescot**
## Summary
- Add an interactive credential mapping flow for unresolved credentials during `n8nac promote`.
- Persist confirmed credential bindings as `<name>:<type>` mappings in promotion config v2 while preserving existing id-based binding resolution.
- Add `--no-interactive` and unit coverage for wizard and CI/blocking behavior.

## Tests
- `npx vitest run packages/cli/tests/unit/promote-command.test.ts`
- `npm run build --workspace=n8nac`

Closes #490

## Summary by CodeRabbit

* **New Features**
  * Interactive credential remapping during promotion—prompt to map unresolved source credentials to target credentials; can be disabled with a new flag.
  * Promotion now persists credential bindings using type-aware binding keys (e.g., "name:type") for more reliable resolution.
* **Behavior Changes**
  * Promotion config is saved only when interactive mappings are confirmed; never saved during dry run. Promotion still blocks when unresolved problems remain.
* **Tests**
  * Added tests covering interactive mapping, duplicate names, and non-interactive blocking.

[Review Change Stack](https://app.coderabbit.ai/change-stack/EtienneLescot/n8n-as-code/pull/491?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: packages/cli/src/commands/promote.ts [hunk 1]
```diff
@@ -542,6 +593,137 @@ export class PromoteCommand {
...
+                routeKey,
+                targetEnvironmentName: target.environmentName,
+            });
+            if (answer.action === 'abort') return 'aborted';
+            if (answer.action === 'skip') continue;
+
+            const targetName = String(answer.credential.name ?? '').trim();
+            const targetType = String(answer.credential.type ?? '').trim();
+            if (!targetName || targetType !== candidate.credentialType) continue;
+            route.bindings!.credentials![this.credentialBindingKey(candidate.sourceName, candidate.credentialType)] = this.credentialBindingKey(targetName, targetType);
```
[the remark this answers is not in the document]
Comment (EtienneLescot) [reply packages/cli/src/commands/promote.ts]: Fixed in 305abaa. Interactive credential mappings now persist the selected target credential id when available, and remapping preserves an existing explicit binding instead of overwriting it during the re-plan. I also added a regression test covering duplicate target credentials with the same name/type.

File: packages/cli/tests/unit/promote-command.test.ts [hunk 2]
```diff
@@ -316,7 +316,151 @@ export class CredentialWorkflow {
...
+                }),
+                promptCredentialMapping: async () => ({ action: 'abort' }),
+            }).run(sourcePath, {
+                from: 'Dev',
+                to: 'Prod',
+                push: false,
+                interactive: false,
+                promotionConfig: promotionConfigPath,
+            })).rejects.toThrow('Promotion blocked by 1 problem.');
+            expect(existsSync(promotionConfigPath)).toBe(false);
```
[the remark this answers is not in the document]
Comment (EtienneLescot) [reply packages/cli/tests/unit/promote-command.test.ts:536]: Fixed in 305abaa. The non-interactive test prompt hook now throws if called, so the test fails fast if `interactive: false` ever invokes `promptCredentialMapping`.
